src/Stryker.Core/Stryker.Core/Mutants/MutantOrchestrator.cs

```diff
@@ -76,72 +216,66 @@ public SyntaxNode Mutate(SyntaxNode currentNode)
-                        currentNodeCopy = currentNodeCopy.ReplaceNode(currentExpressionSyntax, MutateWithConditionalExpressions(currentExpressionSyntax));
+                        var mutant = Mutate(expressionSyntax);
+                        currentNodeCopy = currentNodeCopy.ReplaceNode(currentExpressionSyntax, mutant);
                     }
-                }
-                return currentNodeCopy;
-            }
-
-            if (currentNode is StatementSyntax statement && currentNode.Kind() != SyntaxKind.Block)
-            {
-                // Expression kinds that contain a body and can't have their scope changed should be mutated recursively
-                if (currentNode is LocalFunctionStatementSyntax localFunction)
-                {
-                    return localFunction.ReplaceNode(localFunction.Body, Mutate(localFunction.Body));
-                }
-                if (currentNode is IfStatementSyntax ifStatement)
-                {
-                    if (!ifStatement.Condition.DescendantNodes().Any(x => x.IsKind(SyntaxKind.DeclarationExpression) || x.IsKind(SyntaxKind.DeclarationPattern)))
+                    else if (expressionSyntax is MemberAccessExpressionSyntax memberAccess)
                     {
-                        ifStatement = ifStatement.ReplaceNode(ifStatement.Condition, MutateWithConditionalExpressions(ifStatement.Condition));
+                        actualMutationNode = memberAccess.Expression;
                     }
-                    if (ifStatement.Else != null)
+                    else if (expressionSyntax is AnonymousFunctionExpressionSyntax anonymousFunction)
                     {
-                        ifStatement = ifStatement.ReplaceNode(ifStatement.Else, Mutate(ifStatement.Else));
+                        actualMutationNode = anonymousFunction.Body;
                     }
-                    return ifStatement.ReplaceNode(ifStatement.Statement, Mutate(ifStatement.Statement));
-                }
-                return MutateWithIfStatements(statement);
-            }
-            else
-            {
-                // No statement found yet, search deeper in the tree for statements to mutate
-                var children = currentNode.ChildNodes().ToList();
-                var childCopy = currentNode.TrackNodes(children);
-                foreach (var child in children)
-                {
-                    var mutatedNode = Mutate(child);
-                    var originalNode = childCopy.GetCurrentNode(child);
-                    if (!mutatedNode.IsEquivalentTo(originalNode))
+
+                    if (actualMutationNode != null)
                     {
-                        childCopy = childCopy.ReplaceNode(originalNode, mutatedNode);
+                        var subNodeCopy = expressionSyntax.TrackNodes(actualMutationNode);
+                        var mutant = Mutate(actualMutationNode);
```

Comment:
These local variable names are very unclear. Mutant has a different meaning in the context of mutation testing. This seems off here.

Reply:
Naming is hard 😉. Got rid of some intermediate steps, use _mutationCandidate_ instead of _actualMutationNode_. Sounds better to me